src/app/src/pages/Draw.js

```diff
@@ -1,12 +1,67 @@
+import { useSelector } from 'react-redux';
+import { useParams } from 'react-router-dom';
+import { Box, Flex, Spinner } from '@chakra-ui/react';
+
+import NotFound from './NotFound';
 import DrawTools from '../components/DrawTools';
 import Layers from '../components/Layers';
 import Map from '../components/Map';
+import Sidebar from '../components/Sidebar';
+
+import { useGetBoundaryDetailsQuery } from '../api/boundaries';
+import { BOUNDARY_STATUS, ROLES } from '../constants';
+
+const DRAW_MODES = {
+    FULLY_EDITABLE: 'fully_editable',
+    ANNOTATIONS_ONLY: 'annotations_only',
+    READ_ONLY: 'read_only',
+};
 
 export default function Draw() {
+    const user = useSelector(state => state.auth.user);
+    const { id } = useParams();
+
+    const { isFetching, data: details, error } = useGetBoundaryDetailsQuery(id);
+
+    if (isFetching) {
+        return (
+            <Box w='100%' h='100vh'>
+                <Flex direction='column' alignItems='center'>
+                    <Spinner mt={60} />
+                </Flex>
+            </Box>
+        );
+    }
+
+    if (error) {
```

Comment:
```suggestion
    if (error || typeof details !== 'object') {
```

This will prevent errors from pages like http://localhost:4545/draw/xyz, which send back an HTML response instead of JSON. This is because a lot of 404s are handled by [Django SPA Middleware](https://django-spa.readthedocs.io/en/latest/readme.html) that sends all unknown URLs to the front-end to be resolved by the client router.